app/src/main/java/com/chooloo/www/callmanager/cursorloader/ContactsCursorLoader.java

```diff
@@ -29,7 +29,7 @@
 public class ContactsCursorLoader extends CursorLoader {
 
     // Columns
-    public static String COLUMN_ID = Contacts._ID;
+    public static String COLUMN_ID = Phone.CONTACT_ID;
```

Comment:
android has a fairly complex system to maintain contacts. Here's what I know so far:

- `RawContacts` table stores the actual contacts
- `Data` table stores contact data
- `Contacts` table stores contacts after aggregation of `RawContacts` items using `Data` items (see [here][1])

These tables are connected via keys:

```
RawContacts.CONTACT_ID => Contacts._ID
Data.RAW_CONTACT_ID => RawContacts._ID
Data.CONTACT_ID => Contacts._ID
```

`CommonDataKinds.Phone` is a utility class for data representing a telephone number and `CommonDataKinds.Phone` references original contact's `Contacts._ID`.

Since `ContactsCursorLoader` queries on `Phone.CONTENT_URI` instead of `Contacts.CONTENT_URI`, using `Contacts._ID` directly returns `Phone._ID` values.

[1]: https://developer.android.com/guide/topics/providers/contacts-provider.html#ContactBasics

Reply:
Got it thanks! 
I didn't know all these stuff and it was really a pain in the ass :)
Now I know